scripts/support/create_ge_images/create_template.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash -e
+#
+# Copyright 2015 Telefónica Investigación y Desarrollo, S.A.U
+#
+# This file is part of FI-Core project.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at:
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For those usages not covered by the Apache version 2.0 License please
+# contact with opensource@tid.es
+#
+
+# See: https://forge.fiware.org/plugins/mediawiki/wiki/testbed/index.php/FIWARE_LAB_Image_Deployement_Guideline
+
+set -e
+
+export SHARED_NETWORK_ID=$(neutron net-list |awk -e '/node-int-net-01/ {print $2}')
+export SECURITY_GROUP_CREATE=sshopen
+export SECURITY_GROUP_TEST=allopen
+export KEYPAIR=createimage
+export CREATESCRIPT=create.sh
+export TESTSCRIPT=test.sh
+export UPLOAD_FILE=data.tgz
+export IP=$(nova floating-ip-list | awk '/^\|[ ]+[0-9]+/ { print $2 }')
+
+
+cd $(dirname $0)
+
+create_template_ubuntu12() {
+  # Parameters: name
+  check_params $*
+  create_template $1 base_ubuntu_12.04 ubuntu 
+}
+
+create_template_ubuntu14() {
+  # Parameters: name
+  check_params $*
+  create_template $1 base_ubuntu_14.04 ubuntu
+}
+
+create_template_debian7() {
+  # Parameters: name
+  check_params $*
+  create_template $1 base_debian_7 debian
+}
+
+create_template_centos6() {
+  # Parameters: name
+  check_params $*
+  create_template $1 base_centos_6 centos
+}
+
+create_template_centos7() {
+  # Parameters: name
+  check_params $*
+  create_template $1 base_centos_7 centos
+}
+
+
+create_template() {
+  name=$1
+  base_image=$2
+  user=$3
+
+  # ask sudo password at start
+  sudo true
+  # Launch de VM using base image
+  echo "launching VM $name $base_image $KEYPAIR $IP m1.small"
+  ID=$(nova boot ${name} --flavor m1.small --key-name $KEYPAIR --security-groups $SECURITY_GROUP_CREATE --image ${base_image} --nic net-id=$SHARED_NETWORK_ID | awk -F\| '$2 ~ /[ \t]+id[ \t]+/ {print $3}')
+  echo "VM id is $ID"
+  sleep 5
+  nova floating-ip-associate $ID $IP
+  # Wait until ssh is ready
+  ussh="ssh -oStrictHostKeyChecking=no"
+  ssh-keygen -f ~/.ssh/known_hosts -R $IP
+  echo "Waiting until ssh is ready..."
+  sleep 30
```

Comment:
Why not? If there is a problem with the connectivity this loop could not finish.

echo "Attempting to SSH to server..."  
index=1

while (( $index <= $maxConnectionAttempts ))  
do  
  $ussh ${user}@${IP}
  case $? in
    (0) echo "${index}> Success"; break ;;
    (*) echo "${index} of ${maxConnectionAttempts}> Bastion SSH server not ready yet, waiting ${sleepSeconds} seconds..." ;;
  esac

  sleep $sleepSeconds
  ((index+=1))
done